module/Motion/Rotation.cpp

```diff
@@ -0,0 +1,269 @@
+
+  //両輪が目標距離に到達するまでループ
+  while(leftSign != 0 || rightSign != 0) {
+    // 残りの移動距離
+    double diffLeftDistance
+        = (targetLeftDistance - Mileage::calculateWheelMileage(measurer.getLeftCount())) * leftSign;
+    double diffRightDistance
+        = (targetRightDistance - Mileage::calculateWheelMileage(measurer.getRightCount()))
+          * rightSign;
+
+    // 目標距離に到達した場合
+    if(diffLeftDistance <= 0) {
+      leftSign = 0;
+    }
+    if(diffRightDistance <= 0) {
+      rightSign = 0;
+    }
+
+    // PWM値 = 残りの走行距離/走行距離 * 指定PWM値(最小値 MIN_PWM)
+    int leftPwm = (diffLeftDistance / targetDistance * pwm >= ROTATE_MIN_PWM)
+                      ? diffLeftDistance / targetDistance * pwm
+                      : ROTATE_MIN_PWM;
+    int rightPwm = (diffRightDistance / targetDistance * pwm >= ROTATE_MIN_PWM)
+                       ? diffRightDistance / targetDistance * pwm
+                       : ROTATE_MIN_PWM;
+    controller.setLeftMotorPwm(abs(leftPwm) * leftSign);
+    controller.setRightMotorPwm(abs(rightPwm) * rightSign);
+
+    controller.sleep();
+  }
+
+  //モータの停止
+  controller.stopMotor();
+}
+
+//設定された角度とPWM値で右タイヤを軸に前方へピボットターンする
+void Rotation::turnForwardRightPivot(int angle, int pwm)
+{
+  // pwm値が0の場合はwarningを出して終了する
+  if(pwm == 0) {
+    printf("\x1b[36m"); /* 文字色をシアンに */
+    printf("warning: The pwm value passed to LineTracer::run is 0\n");
+    printf("\x1b[39m"); /* 文字色をデフォルトに戻す */
+    return;
+  }
+  int leftPwm = pwm;
+  int rightPwm = -1;
+
+  controller.resetMotorCount();
+  double motorCount = 0;
+  double targetMotorCount = calculate(angle);
+
+  while(motorCount <= targetMotorCount) {
+    if(pwm == 0) break;
+
+    double leftCountRate = 1 - (measurer.getLeftCount() / targetMotorCount);
+    leftPwm = pwm * leftCountRate > PIVOT_FRONT_MIN_PWM ? (int)(pwm * leftCountRate)
+                                                        : PIVOT_FRONT_MIN_PWM;
+
+    controller.setLeftMotorPwm(leftPwm);
+    controller.setRightMotorPwm(rightPwm);
+
+    motorCount = (measurer.getLeftCount() + measurer.getRightCount()) / 2;
+
+    controller.sleep();
+  }
+
+  controller.stopMotor();
+}
+
+//設定された角度とPWM値で右タイヤを軸に後方へピボットターンする
+void Rotation::turnBackRightPivot(int angle, int pwm)
+{
+  // pwm値が0の場合はwarningを出して終了する
+  if(pwm == 0) {
+    printf("\x1b[36m"); /* 文字色をシアンに */
+    printf("warning: The pwm value passed to LineTracer::run is 0\n");
+    printf("\x1b[39m"); /* 文字色をデフォルトに戻す */
+    return;
+  }
+  angle = abs(angle);
+  int leftPwm = pwm;
+  int rightPwm = 3;
+  controller.resetMotorCount();
+  double motorCount = 0;
+  double targetMotorCount = calculate(angle);
+
+  while(motorCount <= targetMotorCount) {
+    if(pwm == 0) break;
+
+    double leftCountRate = 1 - (abs(measurer.getLeftCount()) / targetMotorCount);
+    leftPwm = pwm * leftCountRate > PIVOT_FRONT_MIN_PWM ? (int)(pwm * leftCountRate)
+                                                        : PIVOT_FRONT_MIN_PWM;
```

Comment:
248行目と同じようにコメントは入れてあげたほうがいい
特にここら辺は分かりづらいし
```suggestion
    //徐々に速度を遅くする処理
    double leftCountRate = 1 - (abs(measurer.getLeftCount()) / targetMotorCount);
    leftPwm = pwm * leftCountRate > PIVOT_FRONT_MIN_PWM ? (int)(pwm * leftCountRate)
                                                        : PIVOT_FRONT_MIN_PWM;
```